### PR TITLE
Feature | Issue #51 | Restructure Table of Content

### LINKED ISSUE #51: [Feature]: Restructure Table of Content
### Contact Details

_No response_

### Feature Type

Something else!

### Description

- Restructure Table of Content

### PR DESCRIPTION
**Pull request by nabeel-shakeel**
## Description
This PR closes #51

Changes proposed in this pull request:
- Updated table of content

## Checklist
- [x] Make sure you are making a pull request against the **main branch**
- [x] Check the PR title and branch naming structure matched with our [guidelines](../CONTRIBUTING.md) 
- Please check the type of change your PR introduces:
  - [ ] New feature
  - [ ] Bug in content and documentation
  - [x] Fix in content and documentation
  - [ ] Other changes

## Screenshots
Screenshot 2022-10-29 at 2 44 45 PM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -94,10 +94,70 @@ These are the react experts to whom content and resources we are referring
...
+  - [useId](#use-id)
+  - [useImperativeHandle](#use-imperative-handle)
+  - [useInsertionEffect](#use-insertion-effect)
+  - [useLayoutEffect](#use-layout-effect)
+  - [useMemo](#use-memo)
+  - [useReducer](#use-reducer)
+  - [useRef](#use-ref)
+  - [useSyncExternalStore](#use-sync-external-store)
+  - [useTransition](#use-transition)
+- [Routing](#routing)
```
Comment (Sheraz064-ai) [inline README.md:113]: This heading is not bold-styled.

File: README.md [hunk 2]
```diff
@@ -94,10 +94,70 @@ These are the react experts to whom content and resources we are referring
...
+  - [React Router](#react-router)
+- [Styling](#styling)
```
Comment (Sheraz064-ai) [inline README.md:115]: Same as above

File: README.md [hunk 3]
```diff
@@ -94,10 +94,70 @@ These are the react experts to whom content and resources we are referring
...
+  - [CSS-in-JS](#css-in-js)
+  - [Inline Styling](#inline-styling)
+  - [Styled Components](#styled-components)
+  - [CSS Modules](#css-modules)
+  - [React Router](#react-router)
+  - [Tailwind CSS](#tailwind-css)
+- [Global State Management](#global-state-management)
```
Comment (Sheraz064-ai) [inline README.md:122]: Same as above

File: README.md [hunk 4]
```diff
@@ -94,10 +94,70 @@ These are the react experts to whom content and resources we are referring
...
+  - [Redux](#redux)
+  - [Recoil](#recoil)
+  - [Jotai](#jotai)
+  - [Rematch](#rematch)
+  - [Hookstate](#hook-state)
+  - [MobX](#mobx)
+  - [Zustand](#zustand)
+- [Data Fetching](#data-fetching)
```
Comment (Sheraz064-ai) [inline README.md:130]: Same as above

File: README.md [hunk 5]
```diff
@@ -94,10 +94,70 @@ These are the react experts to whom content and resources we are referring
...
+  - [Custom Hooks](#custom-hooks)
+  - [Compound Component Pattern](#cmd-comp-pattern)
+  - [Composition Components vs Configuration Components](#cc-vs-cc)
 - **[Testing](#testing)**
-- **[React in TypeScript](#react-in-typescript)**
+  - [React Testing Library](#react-testing-library)
+  - [Jest](#jest)
+  - [Enzyme](#enzyme)
+- **[React in Typescript](#react-in-typescript)**
+- [Server Side Rendering](#server-side-rendering)
```
Comment (Sheraz064-ai) [inline README.md:146]: Same as above

File: README.md [hunk 6]
```diff
@@ -94,10 +94,70 @@ These are the react experts to whom content and resources we are referring
...
+  - [React Dom Server](#react-dom-server)
+  - [NextJS](#next-js)
+  - [Remix](#remix)
+  - [Gatsby](#gatsby)
+- [Security](#security)
```
Comment (Sheraz064-ai) [inline README.md:151]: Same as above

File: README.md [hunk 7]
```diff
@@ -94,10 +94,70 @@ These are the react experts to whom content and resources we are referring
...
+  - [Cyber Attacks](#cyber-attacks)
+  - [Vulnerabilities](#vulnerabilities)
+  - [Best Practices](#sec-best-practices)
+- [Architecture](#architecture)
```
Comment (Sheraz064-ai) [inline README.md:155]: Same as above

File: README.md [hunk 8]
```diff
@@ -94,10 +94,70 @@ These are the react experts to whom content and resources we are referring
...
+- [Toolchains](#toolchains)
```
Comment (Sheraz064-ai) [inline README.md:156]: Same as above

File: README.md [hunk 9]
```diff
@@ -94,10 +94,70 @@ These are the react experts to whom content and resources we are referring
...
+  - [Create React App](#create-react-app)
+  - [Vite](#vite)
+  - [Nx](#nx)
+- [Dockerization](#dockerization)
```
Comment (Sheraz064-ai) [inline README.md:160]: Same as above

**Review by Sheraz064-ai — CHANGES_REQUESTED** [review]
Please fix the requested changes

File: README.md (hunk 1 above)
Comment (nabeel-shakeel) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (nabeel-shakeel) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (nabeel-shakeel) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (nabeel-shakeel) [reply README.md]: Done

File: README.md (hunk 5 above)
Comment (nabeel-shakeel) [reply README.md]: Done

File: README.md (hunk 6 above)
Comment (nabeel-shakeel) [reply README.md]: Done

File: README.md (hunk 7 above)
Comment (nabeel-shakeel) [reply README.md]: Done

File: README.md (hunk 8 above)
Comment (nabeel-shakeel) [reply README.md]: Done

File: README.md (hunk 9 above)
Comment (nabeel-shakeel) [reply README.md]: Done

**Review by Sheraz064-ai — APPROVED** [review]
Ready to ship 🚢

**Pull request merged**